### ISSUE TITLE
 Can more tse (target speaker extraction)  models be implemented?

### ISSUE THREAD

**Issue by KollyYang** [espnet/espnet#5577]
Hi @Emrys365
Thank you very much for adding the code related to TSE, so that we can easily run a TSE training model, But there is currently only  Time-domain SpeakerBeam mode,  Now The most commonly used model for TSE tasks may be the SpEx series,   the performance of TD-SpeakerBeam maybe is not much different from SpEx_plus , But SpEx_plus has more improved models, and many articles are discussing it for comparison. More importantly, SpEx_plus requires Multi-task training, which is very different from espnet TD-SpeakerBeam, but this is also a relatively important function and maybe requires  many changes to the current espnet tse architecture ,  so hope that epsnet will add a SpEx_plus model as a baseline.
thanks!

**Comment by Emrys365** [comment]
Yes, I have two versions of TD-SpeakerBeam, ① one basically following the recipe's default configuration (only one modification in enroll_segment, set to 32000), and ② another using speaker embeddings instead of enrollment audios as the speaker clue. Both have similar performance and training curves.

⬇️Training curve of exp ①
![enrollment](https://github.com/espnet/espnet/assets/18532145/725e6cf4-edb9-486a-8987-1a4b56f8802f)

|dataset|PESQ_NB|STOI|SAR|SDR|SIR|SI_SNR|
|---|---|---|---|---|---|---|
|enhanced_cv_min_8k|3.54|96.41|18.75|18.75|0.00|18.37|
|enhanced_tt_max_8k|3.47|96.60|17.50|17.50|0.00|17.16|
|enhanced_tt_min_8k|3.46|96.35|17.51|17.51|0.00|17.11|

-----

⬇️Training curve of exp ②
![speaker embedding](https://github.com/espnet/espnet/assets/18532145/340b5837-8557-40ae-8580-49045f274a62)

**Comment by KollyYang** [comment]
The results under the causal mode in the video are great. Will this new model also be added to epsnet's tse? thanks

**Comment by Emrys365** [comment]
I guess the data might have some mismatch from my setup. Could you share more details about your data? For example, the sampling rates of audios in `wav.scp`, `spk1.scp`, `enroll_spk1.scp` of each subset in `dump/raw`.

**Comment by Emrys365** [comment]
Can you also share the information about your python environment?

**Basic environments:**
 - OS information: [e.g., Linux 4.9.0-11-amd64 #1 SMP Debian 4.9.189-3+deb9u2 (2019-11-11) x86_64]
 - python version: [e.g. 3.7.3 (default, Mar 27 2019, 22:11:17) [GCC 7.3.0]]
 - espnet version: [e.g. espnet 0.8.0]
 - Git hash [e.g. b88e89fc7246fed4c2842b55baba884fe1b4ecc2]
   - Commit date [e.g. Tue Sep 1 09:32:54 2020 -0400]
 - pytorch version [e.g. pytorch 1.4.0]

You can obtain them by the following command
```
cd <espnet-root>/tools
. ./activate_python.sh

echo "- OS information: `uname -mrsv`"
python3 << EOF
import sys, espnet, torch
pyversion = sys.version.replace('\n', ' ')
print(f"""- python version: \`{pyversion}\`
- espnet version: \`espnet {espnet.__version__}\`
- pytorch version: \`pytorch {torch.__version__}\`""")
EOF
cat << EOF
- Git hash: \`$(git rev-parse HEAD)\`
  - Commit date: \`$(git log -1 --format='%cd')\`
EOF
```

**Environments from `torch.utils.collect_env`:**
e.g.,
```
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: CentOS Linux release 7.5.1804 (Core)
GCC version: (GCC) 7.2.0
CMake version: version 2.8.12.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: TITAN RTX
GPU 1: TITAN RTX
GPU 2: TITAN RTX
GPU 3: TITAN RTX

Nvidia driver version: 440.33.01
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.0
[pip3] torch==1.4.0
[pip3] torch-complex==0.1.1
[pip3] torch-optimizer==0.0.1a14
[pip3] torchaudio==0.4.0
[pip3] warprnnt-pytorch==0.1
[conda] blas                      1.0                         mkl
[conda] mkl                       2020.1                      217
[conda] mkl-service               2.3.0            py37he904b0f_0
[conda] mkl_fft                   1.1.0            py37h23d657b_0
[conda] mkl_random                1.1.1            py37h0573a6f_0
[conda] pytorch                   1.4.0           py3.7_cuda10.0.130_cudnn7.6.3_0    pytorch
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] pytorch-wpe               0.0.0                    pypi_0    pypi
[conda] torch-complex             0.1.1                    pypi_0    pypi
[conda] torch-optimizer           0.0.1a14                 pypi_0    pypi
[conda] torchaudio                0.4.0                    pypi_0    pypi
[conda] warprnnt-pytorch          0.1                      pypi_0    pypi
```
You can obtain them by the following command

```
cd <espnet-root>/tools
. ./activate_python.sh
python3 -m torch.utils.collect_env
```

**Comment by Emrys365** [comment]
Another thing you may try is running stage 8 in `run.sh` to obtain the metrics for the mixture data (`dump/raw/RESULTS.md`) so that we can check if they match the numbers below:
|dataset|STOI|SAR|SDR|SIR|SI_SNR|
|---|---|---|---|---|---|
|cv_min_8k|71.79|0.16|0.16|0.00|-0.00|
|tt_min_8k|73.80|0.15|0.15|0.00|-0.00|

**Comment by Emrys365** [comment]
> > Another thing you may try is running stage 8 in `run.sh` to obtain the metrics for the mixture data (`dump/raw/RESULTS.md`) so that we can check if they match the numbers below:
> > dataset	STOI	SAR	SDR	SIR	SI_SNR
> > cv_min_8k	71.79	0.16	0.16	0.00	-0.00
> > tt_min_8k	73.80	0.15	0.15	0.00	-0.00
> 
> 这个没有训练完成都是可以吗？目前都是训练了48个epoch

Yes, you should be able to run stage 8 if you also pass the argument `--inference_model "valid.loss.best.pth"` to `run.sh`.

**Comment by Emrys365** [comment]
I will take a closer look at the recipe and your setup. I will try to see if I reproduce your problem.
